recipe/build.sh

```diff
@@ -1,20 +1,47 @@
 #!/bin/bash
 
-export LD_LIBRARY_PATH=$(pwd)
-
 # world-writable files are not allowed
-chmod -R o-w $SRC_DIR
+chmod -R o-w "${SRC_DIR}"
 
-# Give install_name_tool enough room to work its magic
-if [ `uname -s` == "Darwin" ]; then
-    export LDFLAGS="${LDFLAGS} -Wl,-headerpad_max_install_names"
+declare -a _config_args
+_config_args+=(-Dprefix="${PREFIX}")
+_config_args+=(-Dusethreads)
+_config_args+=(-Duserelocatableinc)
+_config_args+=(-Dcccdlflags="-fPIC")
+_config_args+=(-Dldflags="${LDFLAGS}")
+# .. ran into too many problems with '.' not being on @INC:
+_config_args+=(-Ddefault_inc_excludes_dot=n)
+if [[ -n "${GCC:-${CC}}" ]]; then
+  _config_args+=("-Dcc=${GCC:-${CC}}")
+fi
+if [[ ${HOST} =~ .*linux.* ]]; then
+  _config_args+=(-Dlddlflags="-shared ${LDFLAGS}")
+# elif [[ ${HOST} =~ .*darwin.* ]]; then
+#   _config_args+=(-Dlddlflags=" -bundle -undefined dynamic_lookup ${LDFLAGS}")
+fi
+# -Dsysroot prevents Configure rummaging around in /usr and
+# linking to system libraries (like GDBM, which is GPL). An
+# alternative is to pass -Dusecrosscompile but that prevents
+# all Configure/run checks which we also do not want.
```

Comment:
Looks like this is causing problems with the compiler stack on conda-forge. @jakirkham @msarahan could you help us here?

Reply:
@mbargull any idea how to best proceed? Shall we try to use `-Dusecrosscompile` or is there a way to make the conda-forge compiler stack compatible with sysroot.

Reply:
I have no idea what the implications are, TBH. But no, I don't think we can use the old compilers for that -- but I don't really know what I'm doing here :see_no_evil:.
Looking at the commit history it seems like @mingwandroid had to handle those intricacies -- Ray, you can tell us what we'll be breaking by omitting `-Dsysroot`?
If it's breaking something non-negligible, we might have to wait until cf uses newer compilers..